website/docs/concepts/router.md

```diff
@@ -164,12 +126,39 @@ fn switch(routes: &Route) -> Html {
 }
 ```
 
-Linking to a specific post is as easy as passing the variant to `Link`:
+:::note 
+You can have a normal `Post` variant instead of `Post {id: String}` too. For example when `Post` is rendered
+with another router, the field can then be redundant as the other router is able to match and handle the path. See the
+[Nested Router](#nested-router) section below for details
+:::
+
+Note the fields must implement `Clone + PartialEq` as part of the `Route` enum. They must also implement
+`std::fmt::Display` and `std::str::FromStr` for serialization and deserialization. Primitive types like integer, float,
+and String already satisfy the requirements.
+
```

Comment:
```suggestion
```
I don't think this note is needed, as you will get a build error if you don't implement them.

Reply:
I agree with hamza. Especially when I first used this without looking at the code. My initial guess was that there was serde magic behind. Gotta say `FromStr` and `Display` are unusual choices compared to `Serialize` and `Deserialize`